go/integration-tests/integration_test.go

```diff
@@ -80,19 +82,10 @@ var Validators = map[string]func(t *testing.T, propertyName string, expected, ac
 				errors = append(errors, ValidationError{Item: i, Property: propertyName, Message: "missing expected property", Expected: expectedPropertyValue, Actual: nil})
 				continue
 			}
-			patch, err := jsondiff.Compare(expectedPropertyValue, actualPropertyValue)
+
+			validationError, err := validateJsonEquality(t, i, propertyName, expectedPropertyValue, actualPropertyValue)
 			if err != nil {
```

Comment:
Should the below code be for when `validationError != nil`, and another case for `err != nil`?